packages/datadog-instrumentations/src/prisma.js

```diff
@@ -0,0 +1,142 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+
+const primsaClientStartCH = channel('apm:prisma:client:start')
+const primsaClientFinishCH = channel('apm:prisma:client:finish')
+const prismaClientErrorCH = channel('apm:prisma:client:error')
+
+const prismaEngineStart = channel('apm:prisma:engine:start')
+
+const allowedClientSpanOperations = [
+  'operation',
+  'serialize',
+  'transaction'
+]
+
+class TracingHelper {
+  dbConfig = null
+  isEnabled () {
+    return true
+  }
+
+  // needs a sampled tracecontext to generate engine spans
+  getTraceParent (context) {
+    return '00-aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa-bbbbbbbbbbbbbbbb-01' // valid sampled traceparent
+  }
+
+  dispatchEngineSpans (spans) {
+    for (const span of spans) {
+      if (span.parentId === null) {
+        prismaEngineStart.publish({ engineSpan: span, allEngineSpans: spans, dbConfig: this.dbConfig })
+      }
+    }
+  }
+
+  getActiveContext () {}
+
+  runInChildSpan (options, callback) {
+    if (typeof options === 'string') {
+      options = {
+        name: options
+      }
+    }
+
+    if (allowedClientSpanOperations.includes(options.name)) {
+      const asyncResource = new AsyncResource('bound-anonymous-fn')
+      const ctx = {
+        resourceName: options.name,
+        attributes: options.attributes || {}
+      }
+
+      return asyncResource.runInAsyncScope(() => {
```

Comment:
`TracingChannel` should be used instead of `AsyncResource`.

Reply:
@rochdev that is definitely true. Should we nevertheless already land this with AsyncResource for now? It would likely otherwise take even more time to get it in? It would also be deactivated by default.

Reply:
The main issue is that there is a lot of tracing code in the publisher. This means that a lot of the code will have to move around anyway. At that point, switching to `TracingChannel` at the same time won't be a ton of effort.

Reply:
I just pushed the change for `TracingChannels` so we should be good here.